### ISSUE TITLE
📝 Svelte: file with two `<script>` blocks, where a non-first block contains type syntax, is silently left unformatted

### ISSUE THREAD

**Issue by efiand** [biomejs/biome#11361]
### Environment information

`@biomejs/biome` 2.5.8 (latest).

Minimal repro:

```svelte
<script lang="ts" module>
let c = 0;
</script>

<script lang="ts">
let { a }: Props = $props()
</script>

<div>{a}</div>
```

Expected: `biome format` applies formatting (adds `;`, normalizes indentation).

Actual: the file is printed unchanged — exit code 0, no diagnostics. Same for `biome check`. The IDE extension behaves identically, so it goes unnoticed.

**Important:** the bug does **not** reproduce in the Playground (WASM build) — the file formats correctly there. It reproduces in the CLI (native binary, v2.5.8) and in the VS Code extension. This suggests a discrepancy between the native and WASM builds of the Svelte parser/formatter.

Narrowing (all in CLI / VS Code):
- With a single `<script>` block the file formats fine.
- `let { a } = $props()` (no type annotation) in the second block formats fine.
- The trigger is any TS type syntax in a non-first script block: `let a: T = foo()`, `let { a } = foo() as T`, `function f(p: T) {}` all reproduce.
- A second `<script>` after a *plain* first `<script>` is fragile even without type syntax: `let a = 1` in the second block also stays unformatted.

Secondary finding in the same file: an `@starting-style` at-rule inside `<style>` makes the **entire** CSS block silently unformatted (the script part still formats).

### Configuration

```json
{
	"$schema": "https://biomejs.dev/schemas/2.5.8/schema.json",
	"assist": {
		"actions": {
			"source": {
				"organizeImports": {
					"level": "on",
					"options": {
						"groups": [
							{
								"source": [":BUN:", ":NODE:", ":PACKAGE_WITH_PROTOCOL:", ":PACKAGE:"],
								"type": false
							},
							":BLANK_LINE:",
							{
								"source": [":ALIAS:", ":PATH:"],
								"type": false
							},
							":BLANK_LINE:",
							{ "type": true }
						]
					}
				},
				"useSortedKeys": "on"
			}
		},
		"enabled": true
	},
	"files": {
		"ignoreUnknown": true,
		"includes": ["**", "!dist", "!node_modules", "!public", "!tmp"]
	},
	"formatter": {
		"enabled": true,
		"indentStyle": "tab",
		"lineWidth": 120
	},
	"javascript": {
		"formatter": {
			"quoteStyle": "single"
		}
	},
	"json": {
		"formatter": {
			"enabled": true
		}
	},
	"linter": {
		"enabled": true,
		"rules": {
			"a11y": "off",
			"complexity": "off",
			"correctness": "off",
			"nursery": "off",
			"performance": "off",
			"security": "off",
			"style": {
				"recommended": false,
				"useExportsLast": "error"
			},
			"suspicious": "off"
		}
	},
	"overrides": [
		{
			"assist": {
				"actions": {
					"source": {
						"useSortedKeys": "off"
					}
				}
			},
			"includes": ["postcss.config.js"]
		}
	],
	"vcs": {
		"clientKind": "git",
		"enabled": true,
		"useIgnoreFile": true
	}
}

```

### Playground link

https://biomejs.dev/playground/?language=svelte&code=PABzAGMAcgBpAHAAdAAgAGwAYQBuAGcAPQAiAHQAcwAiACAAbQBvAGQAdQBsAGUAPgAKAGwAZQB0ACAAYwAgAD0AIAAwADsACgA8AC8AcwBjAHIAaQBwAHQAPgAKAAoAPABzAGMAcgBpAHAAdAAgAGwAYQBuAGcAPQAiAHQAcwAiAD4ACgBsAGUAdAAgAHsAIABhACAAfQA6ACAAUAByAG8AcABzACAAPQAgACQAcAByAG8AcABzACgAKQAKADwALwBzAGMAcgBpAHAAdAA%2BAAoACgA8AGQAaQB2AD4AewBhAH0APAAvAGQAaQB2AD4ACgA%3D

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
That's expected because you're using the first, premature version of HTML-ish files.

You need to enable https://biomejs.dev/reference/configuration/#htmlexperimentalfullsupportenabled for a better UX and support your use case

**Comment by efiand** [comment]
Thanks for the pointer — enabling `html.experimentalFullSupportEnabled` fixed it. Confirmed all three cases from the report now format correctly: the two-`<script>` repro (TS type syntax in the non-first script), a second plain `<script>` block, and the `@starting-style` block inside `<style>` (which I'd assumed was a separate CSS-parser bug).

One gotcha worth documenting: with the flag on, the whole `.svelte` file is formatted through the HTML formatter, so `html.formatter.enabled` must be enabled too — otherwise `biome format` reports "The content was not formatted because the formatter is currently disabled." The docs mention this only for plain `.html` files ("To format HTML files, you also need to enable `html.formatter.enabled`"), so it's easy to miss for Svelte.

Minor UX feedback: in extract mode (flag off), when a `.svelte` script can't be formatted, Biome fails silently — a warning would have saved a lot of debugging. Not a blocker, just a thought.
